### ISSUE TITLE
How to handle in-flight HTLC?

### ISSUE THREAD

**Issue by optout21** [ldksplicing/ldk-splicing#15]
While a splice is in progress, no new HTLC's are accepted, due to the Stfu mode (quiscence).
But any existing pending HTLC is persisted.
Is there any operation needed on the pending HTLC?

**Comment by wpaulino** [comment]
There will be a new quiescence `ChannelState` variant which will disallow _new_ HTLCs. Existing HTLCs will remain pending (as long as the state machine is not pending, i.e., not missing a `commitment_signed`/`revoke_and_ack`) and will exist in the spliced channel as well. Once the splice is acked, the channel (both old and new) can continue offering/accepting HTLCs while the splice is pending confirmation.

**Comment by optout21** [comment]
I agree @wpaulino . What I'm not sure whether some fields of the existing pending HTLCs have to be changed/adjusted, do to change in channel ID, capacity and funding tx. I hope not, but I need to do more research.

One corner case is when the channel capacity is being reduced (splice out), and there is a pending HTLC that would decrease the balance to negative (e.g.: initial channel cap 100, a pending outgoing payment of 60, a splice out with delta -60 -- not possible). But this should be probably handled when checking for allowed relative splice amount.

**Comment by wpaulino** [comment]
Ah, yes a splice as such should be immediately rejected. Most of the balance checks we enforce when sending out HTLCs should also reflect when splicing.
